### PR TITLE
Add industrial_ci clang support

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@gavanderhoorn could you review this PR? I was able to find a work around since all of the package that I need to omit were submodules. I am going to do another commit that makes clang happy.

**Comment by gavanderhoorn** [comment]
> @gavanderhoorn could you review this PR? I was able to find a work around since all of the package that I need to omit were submodules. I am going to do another commit that makes clang happy.

Sure.

If you could take a look at ros-industrial/industrial_core#219 :)

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
+    - $HOME/.ccache
+
 notifications:
   email:
     recipients:
       - levi.armstrong@swri.org
+
 env:
   global:
-    - ROS_DISTRO=kinetic
```
Comment (gavanderhoorn) [inline .travis.yml]: You could keep this and make the lines in the `matrix` section easier to read.

Compare https://github.com/ros-industrial/industrial_core/blob/7e7d21f5936c4a3a3b9ca666453309aa98507d3b/.travis.yml

File: .travis.yml [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+    - env: ROS_DISTRO=kinetic CLANG_FORMAT_CHECK=file
+      git:
+        submodules: false
+    - env: ROS_DISTRO=kinetic ROS_REPO=ros              NOT_TEST_INSTALL=true
+    - env: ROS_DISTRO=kinetic ROS_REPO=ros-shadow-fixed NOT_TEST_INSTALL=true
+    - env: ROS_DISTRO=melodic ROS_REPO=ros              NOT_TEST_INSTALL=true
+    - env: ROS_DISTRO=melodic ROS_REPO=ros-shadow-fixed NOT_TEST_INSTALL=true
+
 install:
   - git clone https://github.com/ros-industrial/industrial_ci.git .industrial_ci
```
Comment (gavanderhoorn) [inline .travis.yml]: Add ` --quiet --depth=1`.

**Comment by Levi-Armstrong** [comment]
@gavanderhoorn How does it look now?

File: .travis.yml (hunk 2 above)
Comment (Levi-Armstrong) [reply .travis.yml]: Made the change.

File: .travis.yml (hunk 1 above)
Comment (Levi-Armstrong) [reply .travis.yml]: I added ROS_REPO and NOT_TEST_INSTALL to the global env.

**Comment by gavanderhoorn** [comment]
I'm not seeing the changes @Levi-Armstrong ?

**Comment by gavanderhoorn** [comment]
O wait. GH UI got a bit confused by the force-pushes.

**Pull request merged**